app/src/main/res/layout/activity_main.xml

```diff
@@ -1,47 +1,41 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    android:orientation="vertical"
     tools:context=".MainActivity">
 
+    <TextView
+        android:id="@+id/welcomeMessage"
+        android:layout_width="214dp"
+        android:layout_height="99dp"
+        android:layout_marginTop="16dp"
```

Comment:
Remember to use constants (in dimens.xml ), esp. for things like margins.

Reply:
Yes you are right. To be fair this isn't my code but as I put everything into a linear layout it thought it was all new code